### DISCUSSION TITLE
Cannot suppress "Module not found: Can't resolve" warning

### DISCUSSION THREAD

**Discussion by hagealex** [vercel/next.js#11511] [Help]
# Bug report

## Describe the bug
Hello everyone,

in the project I'm working on we require on a file which might exist or not:

```
try {
  projectFonts = require('../../config/colors')
} catch (ex) {
  // we dont need a catch handler here as the projectFonts are optional
}
```

This causes a lot of `Module not found: Can't resolve '../../config/colors' in...` warnings on the console.

I tried to suppress these warnings by adding this line of code to the next.config.js 
`config.stats = {warning: false}`:

```
// imports...

module.exports = withStylus({
  webpack: config => {
    config.plugins = config.plugins || []
    config.stats = {warning: false}

    config.plugins = [
      ...config.plugins,

      // Read the .env file
      new Dotenv({
        path: path.join(__dirname, '.env'),
        systemvars: true,
      }),
    ]

    return config
  },
})
```
## Expected behavior

I expected, that `config.stats = {warning: false}` will prevent the warning from being displayed (as mentioned here https://github.com/webpack/webpack.js.org/issues/147#issuecomment-253736058)

## System information

- OS: Ubuntu
- Version of Next.js: 9.3.2

Thanks for your help!

**Comment by timneutkens** [comment]
Disabling all compilation warnings doesn't seem like the correct approach. You should probably not use `require` and use `import()` instead.

**Comment by LukeHalby** [comment]
Hey, 3 years too late, but I had the same issue when importing `aws-amplify` into my next project and on every render would get 

`Module not found: Can't resolve 'aws-crt' in '<some-file-path>\@aws-sdk\util-user-agent-node\dist-cjs'`. 

Here is the solution I found:

```javascript
webpack: (config) => {
  config.resolve.fallback = { "aws-crt": false };

  return config;
},
```
Obviously just replace `"aws-crt"` with whatever package you are having trouble with 👍

**Reply by lukemorton** [reply]
Thanks!
